File: samples/backend.go/product/product.go
package product

import (
	"encoding/json"
	"fmt"
	"math/rand"
	"time"
)

type Item struct {
	ID     int    `json:"id"`
	Name   string `json:"name"`
	Detail string `json:"detail"`
	Price  int    `json:"price"`
}

func NewItem(id int, name string, detail string, price int) *Item {
	return &Item{
		ID:     id,
		Name:   name,
		Detail: detail,
		Price:  price,
	}
}

// 意図的にカオスを挿入しているので、データが正常に帰らないケースもある
func GetItem(items []*Item) string {
	rand.Seed(time.Now().UnixNano())

	i := rand.Intn(9)
	data, err := json.Marshal(items)
	fmt.Println("i =", i)

	if err != nil || i < 5 {
		for {
			time.Sleep(10 * time.Second)
			panic(err)
		}
	}
	fmt.Println(string(data))
	return string(data)
}

File: samples/backend.java/src/main/java/com/example/back/Product.java
package com.example.back;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

import lombok.Data;

@Data
@Entity
public class Product {

    private @Id @GeneratedValue Long id;
    private String name;
    private String detail;
    private int price;

    public Product() {
    }

    public Product(String name, String detail, int price) {

        this.name = name;
        this.detail = detail;
        this.price = price;
    }

}